src/Surfnet/StepupMiddleware/MiddlewareBundle/EventSourcing/ProjectorCollection.php

```diff
@@ -29,9 +30,58 @@
      */
     private $projectors = [];
 
+    /**
+     * @param ProjectorInterface $projector
+     */
     public function add(ProjectorInterface $projector)
     {
-        $this->projectors[] = $projector;
+        $this->projectors[get_class($projector)] = $projector;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getProjectorNames()
+    {
+        return array_map(
+            function (ProjectorInterface $projector) {
+                return get_class($projector);
+            },
+            array_values($this->projectors)
```

Comment:
Let me check, but I thought `array_map` kept the keys as well. That would mean you would have to type out the projector names in full.